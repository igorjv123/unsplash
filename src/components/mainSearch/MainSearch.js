import React, { Component, useState } from 'react'
import TextInput from '../textInput/TextInput'
import './style.sass'
import { connect } from 'react-redux';
import MainInputButton from '../mainInputButton/MainInputButton'
import { func } from 'prop-types'
import setQuery from '../../logic/mainSearch/actions'


const MainSearch = (props) => {

    const [query, setQuery] = useState(props.query)


    const handleInputChange = ({ target }) => {
        setQuery(target.value);
    }

    const handleBtnClick = () => {
        props.setNewQuery(query)
    }

    return(
        <div className = 'header' >
            <TextInput
                className='header__input'
                placeholder={'Type smth...'}
                value = {query}
                onChange={handleInputChange}
            />
            <MainInputButton
                className={'header__search-button'}
                onClick={handleBtnClick}
            >
                Search
            </MainInputButton>
        </div>
    )
}

MainSearch.propTypes = {
    setNewQuery: func
}

MainSearch.defaultProps = {
    setNewQuery: () => {}
}

const mapStateToProps = (state) => {
    const { query } = state;
    return { query }
}


const mapDispatchToProps = (dispatch) => {
    return{
        setNewQuery(query){
            dispatch(setQuery(query))
        }
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(MainSearch);
