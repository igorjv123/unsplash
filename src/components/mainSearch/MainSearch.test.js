import React, {useState} from 'react';
import { MainSearch } from "./MainSearch";
import Enzyme, { mount, shallow, render} from 'enzyme';

describe('MainSearch ', () => {
    // const setStateMock = jest.fn();
    // let useStateSpy = jest.spyOn(React, 'useState')
    // useStateSpy.mockImplementation((init) => [init, setStateMock]);

    const mockSetState = jest.fn();
    //
    jest.mock('react', () => ({
        useState: initial => [initial, mockSetState]
    }));
    // useState = initial => [initial, mockSetState]

    let wrapper;
    const queryValueMock = 'query';
    const setNewQueryMock = jest.fn();
    const historyMock = {
        push: jest.fn()
    }

    beforeEach(function () {
        wrapper = shallow(<MainSearch setNewQuery={setNewQueryMock} history={historyMock} />);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call setNewQuery prop on search button click', () => {
        const button = wrapper.find('.btn-search')

        button.simulate('click');

        expect(setNewQueryMock).toHaveBeenCalled();
    });

    it('should call histoty.push() on categories click ', () =>{
        const tags = wrapper.find('.header__search-button')
        tags.last().simulate('click')

        expect(historyMock.push).toHaveBeenCalled()

    });

    it('should set query in state on input change ', () =>{
        wrapper = mount(<MainSearch setNewQuery={setNewQueryMock} history={historyMock} />);

        const input = wrapper.find('input')
        input.simulate('change', {target: {value: queryValueMock}});

        expect(mockSetState).toHaveBeenCalledWith(queryValueMock);
    });
});