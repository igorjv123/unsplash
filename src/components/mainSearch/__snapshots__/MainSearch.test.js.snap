// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch  should render without crashing 1`] = `
<div
  className="header"
>
  <TextInput
    className="header__input"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Type smth..."
    value=""
  />
  <MainInputButton
    className="header__search-button btn-search"
    onClick={[Function]}
  >
    Search
  </MainInputButton>
  <MainInputButton
    className="header__search-button"
    onClick={[Function]}
  >
    Categories
  </MainInputButton>
</div>
`;
